roofit/roofitcore/inc/TestStatistics/optional_parameter_types.h

```diff
@@ -0,0 +1,48 @@
+// Author: Patrick Bos, Netherlands eScience Center / NIKHEF 2021
+
+/*****************************************************************************
+ * RooFit
+ * Authors:                                                                  *
+ *   WV, Wouter Verkerke, UC Santa Barbara, verkerke@slac.stanford.edu       *
+ *   DK, David Kirkby,    UC Irvine,         dkirkby@uci.edu                 *
+ *                                                                           *
+ * Copyright (c) 2000-2021, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+
+#ifndef ROOT_ROOFIT_TESTSTATISTICS_optional_parameter_types
+#define ROOT_ROOFIT_TESTSTATISTICS_optional_parameter_types
+
+#include <RooArgSet.h>
+
+namespace RooFit {
+namespace TestStatistics {
+
+// strongly named container types for use as optional parameters in the test statistics constructors
+
+struct ConstrainedParameters {
+   ConstrainedParameters() = default;
+   explicit ConstrainedParameters(const RooArgSet &parameters);
+   RooArgSet set;
+};
+
+struct ExternalConstraints {
+   ExternalConstraints() = default;
+   explicit ExternalConstraints(const RooArgSet &constraints);
+   RooArgSet set;
+};
+
+struct GlobalObservables {
+   GlobalObservables() = default;
+   explicit GlobalObservables(const RooArgSet &observables);
+   RooArgSet set;
+};
+
+}
```

Comment:
Are these user-facing helpers? If no, maybe better hide in a `detail` namespace or similar. It's also not nice that they are undocumented.

Reply:
Yes, they are supposed to be user facing. However, we should discuss whether to replace them with Jonas' new test statistic configuration struct.